### PR TITLE
Auto tagger release v2.2 

### PR DESCRIPTION
**Pull request by cpefatimaabdillahi**
Hi @petrjasek,

As our last Pull Request had to be rolled back due to too many errors and insufficient time to troubleshoot between UAT and Production releases, we prepared this ahead of your next UAT release. Most of the changes are carried over from our previous PR, with an additional 3-4 items related to the NINJSV3 formatter. 

The changes include:

**Client**:
1. Translation in the `fr_CA.po` file
2. Replace ampersands in headline, slugline, and extended headline
  - Introduced new variables: `safeSlugline`, `safeHeadline`, `safeHeadlineExtended`
3. Replace "abstract" with "headline_extended"
4. Add `creator` (string) and `relevance` (number)
5. Pass story language to the server from the client
6. Add `aria-label` to HTML elements
7. Enable text wrapping in the search drop-down for longer terms

**Server**:
1. Add article language to URLs
2. Add relevance and creator to tags
3. Add slugline, guid, env, and datetime to the semaphore XML payload
4. Fix `format_relevance` in `semaphore.py`
5. Replaced the old XML transformation logic with a more robust implementation in `semaphore.py`
6. implement propagation of the highest relevance score from child tags to their parent tags ``semaphore.py``

**Environment:**
1. Replace `search_parent_url` with the new URL
 
**Changes needed from Source Fabric as these files are outside of the superdesk-cp scope:**
1. Please update the `scripts/core/superdesk-api.d.ts` file in https://github.com/superdesk/superdesk-client-core/blob/e7409f91a0851f4efce9bb44412ad350a4e01cb3/scripts/core/superdesk-api.d.ts#L3703 so that we can add relevance and creator to `ISubjects`. This allows tags added via search using `getExistingTags` in https://github.com/canadianpress/superdesk-cp-2/blob/version20.11/client/extensions/auto-tagger/src/data-transformations.ts to return relevance and creator.
3. For accessibility reasons, please rename "button" on the auto tagger widget to PIN / UNPIN.
- The Auto-tagger widget currently has a button element without a value text. As a result, screen readers only announce it as a "button," which does not provide users with any context about its functionality. This makes it difficult for users relying on screen readers. Can we name the button Alt text as "Pin" or "Unpin"? Please see the screenshot.

### CODE REVIEW

File: client/extensions/auto-tagger/src/auto-tagging.tsx [hunk 1]
```diff
@@ -587,20 +587,38 @@ export function getAutoTaggingComponent(superdesk: ISuperdesk, label: string): I
...
-                                                                }
-
-                                                                {
-                                                                    _item?.description == null ? null : (
-                                                                        <p aria-label={`Description: ${_item.description}`}>{_item.description}</p>
-                                                                    )
-                                                                }
+                                                            <div 
+                                                                className="auto-tagging-widget__autocomplete-item" 
+                                                                aria-label={`Item name ${_item.name}`}
```
Comment (petrjasek) [inline client/extensions/auto-tagger/src/auto-tagging.tsx:592]: you could also translate those via gettext if needed:
```
aria-label={gettext('Item name {{name}}', {name: _item.name})}
```

**Review by petrjasek — APPROVED** [review]
looks good, will make a PR with those client core changes

**Comment by cpefatimaabdillahi** [comment]
Thanks @petrjasek , please let us know if you have any questions regarding the core changes that were requested.

**Comment by petrjasek** [comment]
can you pls check that PR @cpefatimaabdillahi if I got it right? https://github.com/superdesk/superdesk-client-core/pull/4599

**Comment by cpefatimaabdillahi** [comment]
> can you pls check that PR @cpefatimaabdillahi if I got it right? [superdesk/superdesk-client-core#4599](https://github.com/superdesk/superdesk-client-core/pull/4599)

Hi @petrjasek , the commits look fine however just an fyi, relevance and creator are not specific to AI subjects since even manually added tags/indexes will have both attributes with creator = "human" and a predetermined number for relevance.

**Comment by petrjasek** [comment]
seems like you can do this ISubject change in the extension @cpefatimaabdillahi :
https://github.com/superdesk/superdesk-client-core/pull/4599#discussion_r1716829432

**Comment by cmuldur** [comment]
> seems like you can do this ISubject change in the extension @cpefatimaabdillahi : [superdesk/superdesk-client-core#4599 (comment)](https://github.com/superdesk/superdesk-client-core/pull/4599#discussion_r1716829432)

Sounds good. Just added. @petrjasek

File: client/extensions/auto-tagger/src/globals.d.ts [hunk 2]
```diff
@@ -17,4 +17,9 @@ declare module 'superdesk-api' {
             }
         };
     }
-}
\ No newline at end of file
+
+    interface ISubject {
+        creator: string;
+        relevance: number;
```
Comment (petrjasek) [inline client/extensions/auto-tagger/src/globals.d.ts]: I think you have to make them optional:
```ts
creator?: string;
relevance?: string;
```

Comment (cmuldur) [reply client/extensions/auto-tagger/src/globals.d.ts]: Implemented @petrjasek

**Review by petrjasek — APPROVED** [review]

**Comment by petrjasek** [comment]
can you add me write permissions to your fork? trying to fix the conflicts but it won't allow me to push there

**Comment by cmuldur** [comment]
> can you add me write permissions to your fork? trying to fix the conflicts but it won't allow me to push there

Just sent a request @petrjasek. Although I can just resolve the conflict as well. 
I'll just accept the development branch version.
Request link: https://github.com/canadianpress/superdesk-cp-2/invitations

**Pull request merged**